src/main/java/uk/ac/ebi/eva/pipeline/io/readers/NonAnnotatedVariantsMongoReader.java

```diff
@@ -37,10 +38,17 @@
             + VariantSourceEntryToDBObjectConverter.STUDYID_FIELD;
 
     /**
-     * @param studyId nullable. If null, the query brings all non-annotated variants in the collection.
-     *  If not null, bring only non-annotated variants from that study.
+     * @param studyId Can be the empty string, meaning to bring all non-annotated variants in the collection.
+     *  If the studyId string is not empty, bring only non-annotated variants from that study. This parameter should
+     *  not be null in any case.
      */
-    public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName, String studyId) {
+    public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName, String studyId)
+            throws JobParametersInvalidException {
+        if (studyId == null) {
+            throw new JobParametersInvalidException("NonAnnotatedVariantsMongoReader needs a non-null studyId " +
```

Comment:
The exception type usually thrown with incorrect arguments is `IllegalArgumentException`.